packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -181,3 +277,240 @@ function isMixedLogicalExpression(node: TSESTree.LogicalExpression): boolean {
+    return true;
+  }
+
+  if (isIdentifier(left) && (isNull(right) || isUndefined(right))) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * This is for cases where we check either undefined or null and fall back to
+ * using type information to ensure that our checks are correct.
+ * example: const foo:? string = 'bar';
+ *          foo !== undefined ? foo : 'a string';
+ * output: foo ?? 'a string'
+ */
+function isFixableImplicitTernary({
+  parserServices,
+  checker,
+  requiredOperator,
+  identifier,
+  node,
+}: {
+  parserServices: ReturnType<typeof util.getParserServices>;
+  checker: ts.TypeChecker;
+  requiredOperator: '!==' | '===';
+  identifier: TSESTree.Identifier | TSESTree.MemberExpression;
+  node: TSESTree.ConditionalExpression;
+}): boolean {
+  if (node.test.type !== AST_NODE_TYPES.BinaryExpression) {
+    return false;
+  }
+  const { left, right, operator } = node.test;
+  if (operator !== requiredOperator) {
+    return false;
+  }
+  const isIdentifier = isEqualIndentierCurry(identifier);
+
+  const i = isIdentifier(left) ? left : isIdentifier(right) ? right : null;
+  if (!i) {
+    return false;
+  }
+
+  const tsNode = parserServices.esTreeNodeToTSNodeMap.get(i);
+  const type = checker.getTypeAtLocation(tsNode);
+  const flags = util.getTypeFlags(type);
+
+  if (flags & (ts.TypeFlags.Any | ts.TypeFlags.Unknown)) {
+    return false;
+  }
+
+  const hasNullType = (flags & ts.TypeFlags.Null) !== 0;
+  const hasUndefinedType = (flags & ts.TypeFlags.Undefined) !== 0;
+
+  if (
+    (hasUndefinedType && hasNullType) ||
+    (!hasUndefinedType && !hasNullType)
+  ) {
+    return false;
+  }
+
+  if (hasNullType && (isNull(right) || isNull(left))) {
+    return true;
+  }
+
+  if (hasUndefinedType && (isUndefined(right) || isUndefined(left))) {
+    return true;
+  }
+
+  return false;
+}
+
+function isEqualIndentierCurry(
+  a: TSESTree.Identifier | TSESTree.MemberExpression,
+) {
+  if (a.type === AST_NODE_TYPES.Identifier) {
+    return function (b: any): boolean {
+      if (a.type !== b.type) {
+        return false;
+      }
+      return !!a.name && !!b.name && a.name === b.name;
+    };
+  }
+  return function (b: any): boolean {
+    if (a.type !== b.type) {
+      return false;
+    }
+    return isEqualMemberExpression(a, b);
+  };
+}
+function isEqualMemberExpression(
+  a: TSESTree.MemberExpression,
+  b: TSESTree.MemberExpression,
+): boolean {
+  return (
+    isEqualMemberExpressionProperty(a.property, b.property) &&
+    isEqualMemberExpressionObject(a.object, b.object)
+  );
+}
+
+function isEqualMemberExpressionProperty(
+  a: TSESTree.MemberExpression['property'],
+  b: TSESTree.MemberExpression['property'],
+): boolean {
+  if (a.type !== b.type) {
+    return false;
+  }
+  if (a.type === AST_NODE_TYPES.ThisExpression) {
+    return true;
+  }
+  if (
+    a.type === AST_NODE_TYPES.Literal ||
+    a.type === AST_NODE_TYPES.Identifier
+  ) {
+    return (
+      // @ts-ignore
```

Comment:
Peeking at the error being suppressed here:

```
Property 'name' does not exist on type 'Identifier | BigIntLiteral | BooleanLiteral | NullLiteral | NumberLiteral | RegExpLiteral | StringLiteral'.
  Property 'name' does not exist on type 'BigIntLiteral'
```

...which is reasonable given `AST_NODE_TYPES.Literal` can be any of those. I suspect you'll want to split this if statement into one if statement for each of the two `a.type` cases.